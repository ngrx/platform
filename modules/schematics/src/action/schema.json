{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxAction",
  "title": "NgRx Action Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the action.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What should be the name of the action?"
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "skipTests": {
      "type": "boolean",
      "description": "When true, does not create test files.",
      "default": false
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "group": {
      "type": "boolean",
      "default": false,
      "description": "Group actions file within 'actions' folder",
      "aliases": ["g"]
    },
    "api": {
      "type": "boolean",
      "default": false,
      "description": "Specifies if api success and failure actions should be generated.",
      "aliases": ["a"],
      "x-prompt": "Should we generate success and failure actions?"
    },
    "creators": {
      "type": "boolean",
      "default": true,
      "description": "Specifies whether to use creator functions for handling actions and reducers.",
      "aliases": ["c"],
      "x-prompt": "Do you want to use the create function?"
    }
  },
  "required": []
}
