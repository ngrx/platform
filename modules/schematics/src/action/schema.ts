export interface Schema {
  /**
   * The name of the component.
   */

  name: string;
  /**
   * The path to create the component.
   */

  path?: string;
  /**
   * The name of the project.
   */
  project?: string;
  /**
   * Specifies if a spec file is generated.
   */
  spec?: boolean;
  /**
   * Flag to indicate if a dir is created.
   */

  flat?: boolean;
  /**
   * Group actions file within 'actions' folder
   */

  group?: boolean;
}
