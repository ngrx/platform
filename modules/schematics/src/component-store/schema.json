{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxComponentStore",
  "title": "NgRx Component Store Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the component store.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What should be the name of the component store?"
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component store.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "skipTests": {
      "type": "boolean",
      "default": false,
      "description": "When true, does not create test files."
    },
    "component": {
      "type": "string",
      "default": "",
      "description": "Allows specification of the declaring component.",
      "alias": "m",
      "subtype": "filepath",
      "x-prompt": "To which component (path) should the component store be provided in?"
    },
    "module": {
      "type": "string",
      "default": "",
      "description": "Allows specification of the declaring module.",
      "alias": "m",
      "subtype": "filepath",
      "x-prompt": "To which module (path) should the component store be provided in?"
    }
  },
  "required": []
}
