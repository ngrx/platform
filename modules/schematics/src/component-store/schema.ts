export interface Schema {
  /**
   * The name of the component store.
   */
  name: string;

  /**
   * The path to create the component store.
   */
  path?: string;

  /**
   * The name of the project.
   */
  project?: string;

  /**
   * Flag to indicate if a dir is created.
   */
  flat?: boolean;

  /**
   * When true, does not create test files.
   */
  skipTests?: boolean;

  /**
   * Allows specification of the declaring component.
   */
  component?: string;

  /**
   * Allows specification of the declaring module.
   */
  module?: string;
}
