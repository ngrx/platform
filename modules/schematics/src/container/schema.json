{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxContainer",
  "title": "NgRx Container Options Schema",
  "type": "object",
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What should be the name of the container component?"
    },
    "inlineStyle": {
      "description": "Specifies if the style will be in the ts file.",
      "type": "boolean",
      "alias": "s"
    },
    "inlineTemplate": {
      "description": "Specifies if the template will be in the ts file.",
      "type": "boolean",
      "alias": "t"
    },
    "viewEncapsulation": {
      "description": "Specifies the view encapsulation strategy.",
      "enum": ["Emulated", "Native", "None"],
      "type": "string",
      "alias": "v"
    },
    "changeDetection": {
      "description": "Specifies the change detection strategy.",
      "enum": ["Default", "OnPush"],
      "type": "string",
      "alias": "c"
    },
    "prefix": {
      "type": "string",
      "format": "html-selector",
      "description": "The prefix to apply to generated selectors.",
      "alias": "p"
    },
    "style": {
      "description": "The file extension or preprocessor to use for style files.",
      "type": "string"
    },
    "skipTests": {
      "type": "boolean",
      "description": "When true, does not create test files."
    },
    "flat": {
      "type": "boolean",
      "description": "Flag to indicate if a dir is created."
    },
    "skipImport": {
      "type": "boolean",
      "description": "Flag to skip the module import."
    },
    "selector": {
      "type": "string",
      "format": "html-selector",
      "description": "The selector to use for the component."
    },
    "module": {
      "type": "string",
      "description": "Allows specification of the declaring module.",
      "alias": "m"
    },
    "export": {
      "type": "boolean",
      "description": "Specifies if declaring module exports the component."
    },
    "state": {
      "type": "string",
      "description": "Specifies the path to the state exports."
    },
    "stateInterface": {
      "type": "string",
      "default": "State",
      "description": "Specifies the interface for the state."
    },
    "testDepth": {
      "description": "Specifies whether to create a unit test or an integration test.",
      "enum": ["unit", "integration"],
      "type": "string",
      "default": "integration"
    }
  },
  "required": []
}
