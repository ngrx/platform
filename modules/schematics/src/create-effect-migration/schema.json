{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxCreateEffectMigration",
  "title": "NgRx Effect Migration from @Effect to createEffect",
  "type": "object",
  "properties": {},
  "required": []
}
