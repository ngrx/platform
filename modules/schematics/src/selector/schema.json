{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxSelector",
  "title": "NgRx Selector Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the selector.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the selectors.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "spec": {
      "type": "boolean",
      "description": "Specifies if a spec file is generated.",
      "default": false
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "group": {
      "type": "boolean",
      "default": false,
      "description": "Group selector file within 'selectors' folder",
      "aliases": ["g"]
    },
    "feature": {
      "type": "boolean",
      "default": false,
      "description": "Flag to indicate if part of a feature schematic.",
      "visible": false
    }
  },
  "required": []
}
