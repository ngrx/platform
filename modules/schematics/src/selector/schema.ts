export interface Schema {
  /**
   * The name of the selector.
   */
  name: string;

  /**
   * The path to create the selector.
   */
  path?: string;

  /**
   * The name of the project.
   */
  project?: string;

  /**
   * Flag to indicate if a dir is created.
   */
  flat?: boolean;

  /**
   * When true, does not create test files.
   */
  skipTests?: boolean;

  /**
   * Specifies if this is grouped within a feature
   */
  feature?: boolean;

  /**
   * Specifies if this is grouped within an 'selectors' folder
   */
  group?: boolean;
}
