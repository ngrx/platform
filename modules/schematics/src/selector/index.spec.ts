import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import * as path from 'path';
import { Schema as SelectorOptions } from './schema';
import {
  getTestProjectPath,
  createWorkspace,
} from '@ngrx/schematics-core/testing';

describe('Selector Schematic', () => {
  const schematicRunner = new SchematicTestRunner(
    '@ngrx/schematics',
    path.join(__dirname, '../../collection.json')
  );
  const defaultOptions: SelectorOptions = {
    name: 'foo',
    project: 'bar',
  };

  const projectPath = getTestProjectPath();

  let appTree: UnitTestTree;

  beforeEach(async () => {
    appTree = await createWorkspace(schematicRunner, appTree);
  });

  it('should create selector files', async () => {
    const tree = await schematicRunner.runSchematic(
      'selector',
      defaultOptions,
      appTree
    );

    const selectorPath = `${projectPath}/src/app/foo.selectors.ts`;
    const specPath = `${projectPath}/src/app/foo.selectors.spec.ts`;

    expect(tree.files.includes(selectorPath)).toBeTruthy();
    expect(tree.files.includes(specPath)).toBeTruthy();

    expect(tree.readContent(selectorPath)).toMatchSnapshot();
    expect(tree.readContent(specPath)).toMatchSnapshot();
  });

  it('should not create a spec file if spec is false', async () => {
    const options = {
      ...defaultOptions,
      skipTests: true,
    };
    const tree = await schematicRunner.runSchematic(
      'selector',
      options,
      appTree
    );

    expect(
      tree.files.includes(`${projectPath}/src/app/foo.selectors.spec.ts`)
    ).toBeFalsy();
  });

  it('should group selectors if group is true', async () => {
    const options = {
      ...defaultOptions,
      group: true,
    };
    const tree = await schematicRunner.runSchematic(
      'selector',
      options,
      appTree
    );

    const selectorPath = `${projectPath}/src/app/selectors/foo.selectors.ts`;
    const specPath = `${projectPath}/src/app/selectors/foo.selectors.spec.ts`;

    expect(tree.files.includes(selectorPath)).toBeTruthy();
    expect(tree.files.includes(specPath)).toBeTruthy();

    expect(tree.readContent(selectorPath)).toMatchSnapshot();
    expect(tree.readContent(specPath)).toMatchSnapshot();
  });

  it('should not flatten selectors if flat is false', async () => {
    const options = {
      ...defaultOptions,
      flat: false,
    };
    const tree = await schematicRunner.runSchematic(
      'selector',
      options,
      appTree
    );

    const selectorPath = `${projectPath}/src/app/foo/foo.selectors.ts`;
    const specPath = `${projectPath}/src/app/foo/foo.selectors.spec.ts`;

    expect(tree.files.includes(selectorPath)).toBeTruthy();
    expect(tree.files.includes(specPath)).toBeTruthy();

    expect(tree.readContent(selectorPath)).toMatchSnapshot();
    expect(tree.readContent(specPath)).toMatchSnapshot();
  });

  describe('With feature flag', () => {
    it('should create a selector', async () => {
      const options = {
        ...defaultOptions,
        feature: true,
      };

      const tree = await schematicRunner.runSchematic(
        'selector',
        options,
        appTree
      );

      const selectorPath = `${projectPath}/src/app/foo.selectors.ts`;
      const specPath = `${projectPath}/src/app/foo.selectors.spec.ts`;

      expect(tree.files.includes(selectorPath)).toBeTruthy();
      expect(tree.files.includes(specPath)).toBeTruthy();

      expect(tree.readContent(selectorPath)).toMatchSnapshot();
      expect(tree.readContent(specPath)).toMatchSnapshot();
    });

    it('should group and nest the selectors within a feature', async () => {
      const options = {
        ...defaultOptions,
        feature: true,
        group: true,
        flat: false,
      };

      const tree = await schematicRunner.runSchematic(
        'selector',
        options,
        appTree
      );
      const selectorPath = `${projectPath}/src/app/selectors/foo/foo.selectors.ts`;
      const specPath = `${projectPath}/src/app/selectors/foo/foo.selectors.spec.ts`;

      expect(tree.files.includes(selectorPath)).toBeTruthy();
      expect(tree.files.includes(specPath)).toBeTruthy();

      expect(tree.readContent(selectorPath)).toMatchSnapshot();
      expect(tree.readContent(specPath)).toMatchSnapshot();
    });
  });
});
