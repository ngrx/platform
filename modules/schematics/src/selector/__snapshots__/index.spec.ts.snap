// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector Schematic With feature flag should create a selector 1`] = `
"import { createFeatureSelector, createSelector } from '@ngrx/store';
import * as fromFoo from './foo.reducer';

export const selectFooState = createFeatureSelector<fromFoo.State>(
  fromFoo.fooFeatureKey
);
"
`;

exports[`Selector Schematic With feature flag should create a selector 2`] = `
"import * as fromFoo from './foo.reducer';
import { selectFooState } from './foo.selectors';

describe('Foo Selectors', () => {
  it('should select the feature state', () => {
    const result = selectFooState({
      [fromFoo.fooFeatureKey]: {}
    });

    expect(result).toEqual({});
  });
});
"
`;

exports[`Selector Schematic With feature flag should group and nest the selectors within a feature 1`] = `
"import { createFeatureSelector, createSelector } from '@ngrx/store';
import * as fromFoo from '../../reducers/foo/foo.reducer';

export const selectFooState = createFeatureSelector<fromFoo.State>(
  fromFoo.fooFeatureKey
);
"
`;

exports[`Selector Schematic With feature flag should group and nest the selectors within a feature 2`] = `
"import * as fromFoo from '../../reducers/foo/foo.reducer';
import { selectFooState } from './foo.selectors';

describe('Foo Selectors', () => {
  it('should select the feature state', () => {
    const result = selectFooState({
      [fromFoo.fooFeatureKey]: {}
    });

    expect(result).toEqual({});
  });
});
"
`;

exports[`Selector Schematic should create selector files 1`] = `
"import { createFeatureSelector, createSelector } from '@ngrx/store';

"
`;

exports[`Selector Schematic should create selector files 2`] = `
"

describe('Foo Selectors', () => {
  it('should select the feature state', () => {
    
  });
});
"
`;

exports[`Selector Schematic should group selectors if group is true 1`] = `
"import { createFeatureSelector, createSelector } from '@ngrx/store';

"
`;

exports[`Selector Schematic should group selectors if group is true 2`] = `
"

describe('Foo Selectors', () => {
  it('should select the feature state', () => {
    
  });
});
"
`;

exports[`Selector Schematic should not flatten selectors if flat is false 1`] = `
"import { createFeatureSelector, createSelector } from '@ngrx/store';

"
`;

exports[`Selector Schematic should not flatten selectors if flat is false 2`] = `
"

describe('Foo Selectors', () => {
  it('should select the feature state', () => {
    
  });
});
"
`;
